--- chromium-33.0.1750.115/build/common.gypi.orig	2014-02-21 11:53:43.916922963 +0100
+++ chromium-33.0.1750.115/build/common.gypi	2014-02-21 12:07:44.203951976 +0100
@@ -48,7 +48,7 @@
           'toolkit_views%': 0,
 
           # Use OpenSSL instead of NSS. Under development: see http://crbug.com/62803
-          'use_openssl%': 0,
+          'use_openssl%': 0,
 
           # Disable viewport meta tag by default.
           'enable_viewport%': 0,
@@ -330,7 +330,7 @@
       'safe_browsing%': 1,
 
       # Speech input is compiled in by default. Set to 0 to disable.
-      'input_speech%': 1,
+      'input_speech%': 1, # disabling breaks build
 
       # Notifications are compiled in by default. Set to 0 to disable.
       'notifications%' : 1,
@@ -397,10 +397,10 @@
       'enable_extensions%': 1,
 
       # Enable browser automation.
-      'enable_automation%': 1,
+      'enable_automation%': 0,
 
       # Enable Google Now.
-      'enable_google_now%': 1,
+      'enable_google_now%': 0,
 
       # Enable printing support and UI. This variable is used to configure
       # which parts of printing will be built. 0 disables printing completely,
@@ -722,7 +722,7 @@
         ['OS=="android" or OS=="ios"', {
           'enable_captive_portal_detection%': 0,
         }, {
-          'enable_captive_portal_detection%': 1,
+          'enable_captive_portal_detection%': 0,
         }],
 
         # Enable Skia UI text drawing incrementally on different platforms.
