diff -pruNd chromium-45.0.2454.15.orig/chrome/browser/extensions/extension_service.cc chromium-45.0.2454.15/chrome/browser/extensions/extension_service.cc
--- chromium-45.0.2454.15.orig/chrome/browser/extensions/extension_service.cc	2015-07-23 22:09:21.000000000 +0200
+++ chromium-45.0.2454.15/chrome/browser/extensions/extension_service.cc	2015-07-29 23:45:12.240864172 +0200
@@ -1111,13 +1111,6 @@ void ExtensionService::NotifyExtensionLo
     ThemeSource* theme_source = new ThemeSource(profile_);
     content::URLDataSource::Add(profile_, theme_source);
   }
-
-  // Same for chrome://thumb/ resources.
-  if (permissions_data->HasHostPermission(
-          GURL(chrome::kChromeUIThumbnailURL))) {
-    ThumbnailSource* thumbnail_source = new ThumbnailSource(profile_, false);
-    content::URLDataSource::Add(profile_, thumbnail_source);
-  }
 }
 
 void ExtensionService::NotifyExtensionUnloaded(
diff -pruNd chromium-45.0.2454.15.orig/chrome/browser/resources/local_ntp/local_ntp.html chromium-45.0.2454.15/chrome/browser/resources/local_ntp/local_ntp.html
--- chromium-45.0.2454.15.orig/chrome/browser/resources/local_ntp/local_ntp.html	2015-07-23 22:09:21.000000000 +0200
+++ chromium-45.0.2454.15/chrome/browser/resources/local_ntp/local_ntp.html	2015-07-29 23:45:12.240864172 +0200
@@ -12,19 +12,6 @@
 </head>
 <body>
   <div id="ntp-contents">
-    <div id="most-visited">
-      <div id="mv-tiles"></div>
-      <!-- Notification shown when a tile is blacklisted. -->
-      <div id="mv-notice" class="mv-notice-hide">
-        <span id="mv-msg"></span>
-        <!-- Links in the notification. -->
-        <span id="mv-notice-links">
-          <span id="mv-undo" tabIndex="1"></span>
-          <span id="mv-restore" tabIndex="1"></span>
-          <div id="mv-notice-x" tabIndex="1"></div>
-        </span>
-      </div>
-    </div>
     <div id="attribution"><div id="attribution-text"></div></div>
   </div>
 </body>
diff -pruNd chromium-45.0.2454.15.orig/chrome/browser/resources/local_ntp/local_ntp.js chromium-45.0.2454.15/chrome/browser/resources/local_ntp/local_ntp.js
--- chromium-45.0.2454.15.orig/chrome/browser/resources/local_ntp/local_ntp.js	2015-07-23 22:09:21.000000000 +0200
+++ chromium-45.0.2454.15/chrome/browser/resources/local_ntp/local_ntp.js	2015-07-29 23:47:15.074201495 +0200
@@ -245,19 +245,19 @@ var fakeboxInputBehavior = NTP_DISPOSE_S
 
 
 /** @type {number} @const */
-var MAX_NUM_TILES_TO_SHOW = 8;
+var MAX_NUM_TILES_TO_SHOW = 0;
 
 
 /** @type {number} @const */
-var MIN_NUM_COLUMNS = 2;
+var MIN_NUM_COLUMNS = 0;
 
 
 /** @type {number} @const */
-var MAX_NUM_COLUMNS = 4;
+var MAX_NUM_COLUMNS = 0;
 
 
 /** @type {number} @const */
-var NUM_ROWS = 2;
+var NUM_ROWS = 0;
 
 
 /**
@@ -792,7 +792,6 @@ function init() {
 
   ntpApiHandle = topLevelHandle.newTabPage;
   ntpApiHandle.onthemechange = onThemeChange;
-  ntpApiHandle.onmostvisitedchange = onMostVisitedChange;
 
   ntpApiHandle.oninputstart = onInputStart;
   ntpApiHandle.oninputcancel = restoreNtp;
@@ -850,11 +849,6 @@ function init() {
     document.documentElement.classList.add(CLASSES.RTL);
   }
 
-  var iframe = document.createElement('iframe');
-  // Change the order of tabbing the page to start with NTP tiles.
-  iframe.setAttribute('tabindex', '1');
-  iframe.id = 'mv-single';
-
   var args = [];
 
   if (searchboxApiHandle.rtl)
@@ -866,16 +860,6 @@ function init() {
 
   args.push('removeTooltip=' +
       encodeURIComponent(configData.translatedStrings.removeThumbnailTooltip));
-
-  iframe.src = '//most-visited/single.html?' + args.join('&');
-  $(IDS.TILES).appendChild(iframe);
-
-  iframe.onload = function() {
-    reloadTiles();
-    renderTheme();
-  };
-
-  window.addEventListener('message', handlePostMessage);
 }
 
 
diff -pruNd chromium-45.0.2454.15.orig/chrome/browser/search/instant_service.cc chromium-45.0.2454.15/chrome/browser/search/instant_service.cc
--- chromium-45.0.2454.15.orig/chrome/browser/search/instant_service.cc	2015-07-23 22:09:21.000000000 +0200
+++ chromium-45.0.2454.15/chrome/browser/search/instant_service.cc	2015-07-29 23:45:12.244197505 +0200
@@ -112,9 +112,6 @@ InstantService::InstantService(Profile*
   // on Android.
 #if !defined(OS_ANDROID)
   content::URLDataSource::Add(profile_, new LocalNtpSource(profile_));
-  content::URLDataSource::Add(profile_, new ThumbnailSource(profile_, false));
-  content::URLDataSource::Add(profile_, new ThumbnailSource(profile_, true));
-  content::URLDataSource::Add(profile_, new ThumbnailListSource(profile_));
 #endif  // !defined(OS_ANDROID)
 
   favicon::FallbackIconService* fallback_icon_service =
@@ -127,9 +124,6 @@ InstantService::InstantService(Profile*
       profile_, new FaviconSource(profile_, FaviconSource::FAVICON));
   content::URLDataSource::Add(
       profile_, new LargeIconSource(fallback_icon_service, large_icon_service));
-  content::URLDataSource::Add(profile_, new MostVisitedIframeSource());
-  content::URLDataSource::Add(
-      profile_, new suggestions::SuggestionsSource(profile_));
 }
 
 InstantService::~InstantService() {
diff -pruNd chromium-45.0.2454.15.orig/chrome/browser/search/search.cc chromium-45.0.2454.15/chrome/browser/search/search.cc
--- chromium-45.0.2454.15.orig/chrome/browser/search/search.cc	2015-07-23 22:09:21.000000000 +0200
+++ chromium-45.0.2454.15/chrome/browser/search/search.cc	2015-07-29 23:45:12.244197505 +0200
@@ -268,7 +268,7 @@ NewTabURLState IsValidNewTabURL(Profile*
     return NEW_TAB_URL_INSECURE;
   if (!IsURLAllowedForSupervisedUser(new_tab_url, profile))
     return NEW_TAB_URL_BLOCKED;
-  return NEW_TAB_URL_VALID;
+  return NEW_TAB_URL_NOT_SET;
 }
 
 // Used to look up the URL to use for the New Tab page. Also tracks how we
diff -pruNd chromium-45.0.2454.15.orig/chrome/browser/ui/search/search_ipc_router_policy_impl.cc chromium-45.0.2454.15/chrome/browser/ui/search/search_ipc_router_policy_impl.cc
--- chromium-45.0.2454.15.orig/chrome/browser/ui/search/search_ipc_router_policy_impl.cc	2015-07-23 22:09:22.000000000 +0200
+++ chromium-45.0.2454.15/chrome/browser/ui/search/search_ipc_router_policy_impl.cc	2015-07-29 23:45:12.244197505 +0200
@@ -23,7 +23,7 @@ SearchIPCRouterPolicyImpl::SearchIPCRout
 SearchIPCRouterPolicyImpl::~SearchIPCRouterPolicyImpl() {}
 
 bool SearchIPCRouterPolicyImpl::ShouldProcessSetVoiceSearchSupport() {
-  return true;
+  return false;
 }
 
 bool SearchIPCRouterPolicyImpl::ShouldProcessFocusOmnibox(bool is_active_tab) {
@@ -35,15 +35,15 @@ bool SearchIPCRouterPolicyImpl::ShouldPr
 }
 
 bool SearchIPCRouterPolicyImpl::ShouldProcessDeleteMostVisitedItem() {
-  return !is_incognito_ && chrome::IsInstantNTP(web_contents_);
+  return false;
 }
 
 bool SearchIPCRouterPolicyImpl::ShouldProcessUndoMostVisitedDeletion() {
-  return !is_incognito_ && chrome::IsInstantNTP(web_contents_);
+  return false;
 }
 
 bool SearchIPCRouterPolicyImpl::ShouldProcessUndoAllMostVisitedDeletions() {
-  return !is_incognito_ && chrome::IsInstantNTP(web_contents_);
+  return false;
 }
 
 bool SearchIPCRouterPolicyImpl::ShouldProcessLogEvent() {
@@ -89,7 +89,7 @@ bool SearchIPCRouterPolicyImpl::ShouldSe
 }
 
 bool SearchIPCRouterPolicyImpl::ShouldSendMostVisitedItems() {
-  return !is_incognito_ && chrome::IsInstantNTP(web_contents_);
+  return false;
 }
 
 bool SearchIPCRouterPolicyImpl::ShouldSendThemeBackgroundInfo() {
@@ -97,7 +97,7 @@ bool SearchIPCRouterPolicyImpl::ShouldSe
 }
 
 bool SearchIPCRouterPolicyImpl::ShouldSendToggleVoiceSearch() {
-  return true;
+  return false;
 }
 
 bool SearchIPCRouterPolicyImpl::ShouldSubmitQuery() {
diff -pruNd chromium-45.0.2454.15.orig/components/history/core/browser/top_sites_backend.cc chromium-45.0.2454.15/components/history/core/browser/top_sites_backend.cc
--- chromium-45.0.2454.15.orig/components/history/core/browser/top_sites_backend.cc	2015-07-23 22:09:26.000000000 +0200
+++ chromium-45.0.2454.15/components/history/core/browser/top_sites_backend.cc	2015-07-29 23:45:12.244197505 +0200
@@ -21,58 +21,36 @@ namespace history {
 
 TopSitesBackend::TopSitesBackend(
     const scoped_refptr<base::SingleThreadTaskRunner>& db_task_runner)
-    : db_(new TopSitesDatabase()), db_task_runner_(db_task_runner) {
+    : db_(static_cast< TopSitesDatabase *> (NULL)), db_task_runner_(db_task_runner) {
   DCHECK(db_task_runner_);
 }
 
 void TopSitesBackend::Init(const base::FilePath& path) {
   db_path_ = path;
-  db_task_runner_->PostTask(
-      FROM_HERE, base::Bind(&TopSitesBackend::InitDBOnDBThread, this, path));
 }
 
 void TopSitesBackend::Shutdown() {
-  db_task_runner_->PostTask(
-      FROM_HERE, base::Bind(&TopSitesBackend::ShutdownDBOnDBThread, this));
 }
 
 void TopSitesBackend::GetMostVisitedThumbnails(
     const GetMostVisitedThumbnailsCallback& callback,
     base::CancelableTaskTracker* tracker) {
-  scoped_refptr<MostVisitedThumbnails> thumbnails = new MostVisitedThumbnails();
-  tracker->PostTaskAndReply(
-      db_task_runner_.get(), FROM_HERE,
-      base::Bind(&TopSitesBackend::GetMostVisitedThumbnailsOnDBThread, this,
-                 thumbnails),
-      base::Bind(callback, thumbnails));
 }
 
 void TopSitesBackend::UpdateTopSites(const TopSitesDelta& delta,
                                      const RecordHistogram record_or_not) {
-  db_task_runner_->PostTask(
-      FROM_HERE,
-      base::Bind(&TopSitesBackend::UpdateTopSitesOnDBThread, this, delta,
-                 record_or_not));
 }
 
 void TopSitesBackend::SetPageThumbnail(const MostVisitedURL& url,
                                        int url_rank,
                                        const Images& thumbnail) {
-  db_task_runner_->PostTask(
-      FROM_HERE, base::Bind(&TopSitesBackend::SetPageThumbnailOnDBThread, this,
-                            url, url_rank, thumbnail));
 }
 
 void TopSitesBackend::ResetDatabase() {
-  db_task_runner_->PostTask(
-      FROM_HERE,
-      base::Bind(&TopSitesBackend::ResetDatabaseOnDBThread, this, db_path_));
 }
 
 void TopSitesBackend::DoEmptyRequest(const base::Closure& reply,
                                      base::CancelableTaskTracker* tracker) {
-  tracker->PostTaskAndReply(db_task_runner_.get(), FROM_HERE,
-                            base::Bind(&base::DoNothing), reply);
 }
 
 TopSitesBackend::~TopSitesBackend() {
