diff -pruN chromium-43.0.2357.18.orig/chrome/browser/extensions/extension_service.cc chromium-43.0.2357.18/chrome/browser/extensions/extension_service.cc
--- chromium-43.0.2357.18.orig/chrome/browser/extensions/extension_service.cc	2015-04-15 05:55:03.000000000 +0200
+++ chromium-43.0.2357.18/chrome/browser/extensions/extension_service.cc	2015-04-23 09:46:06.806900140 +0200
@@ -1100,13 +1100,6 @@ void ExtensionService::NotifyExtensionLo
     ThemeSource* theme_source = new ThemeSource(profile_);
     content::URLDataSource::Add(profile_, theme_source);
   }
-
-  // Same for chrome://thumb/ resources.
-  if (permissions_data->HasHostPermission(
-          GURL(chrome::kChromeUIThumbnailURL))) {
-    ThumbnailSource* thumbnail_source = new ThumbnailSource(profile_, false);
-    content::URLDataSource::Add(profile_, thumbnail_source);
-  }
 }
 
 void ExtensionService::NotifyExtensionUnloaded(
diff -pruN chromium-43.0.2357.18.orig/chrome/browser/resources/local_ntp/local_ntp.html chromium-43.0.2357.18/chrome/browser/resources/local_ntp/local_ntp.html
--- chromium-43.0.2357.18.orig/chrome/browser/resources/local_ntp/local_ntp.html	2015-04-15 05:55:03.000000000 +0200
+++ chromium-43.0.2357.18/chrome/browser/resources/local_ntp/local_ntp.html	2015-04-23 09:46:06.806900140 +0200
@@ -12,19 +12,6 @@
 </head>
 <body>
   <div id="ntp-contents" class="des-mat">
-    <div id="most-visited">
-      <div id="mv-tiles"></div>
-      <!-- Notification shown when a tile is blacklisted. -->
-      <div id="mv-notice" class="mv-notice-hide">
-        <span id="mv-msg"></span>
-        <!-- Links in the notification. -->
-        <span id="mv-notice-links">
-          <span id="mv-undo" tabIndex="1"></span>
-          <span id="mv-restore" tabIndex="1"></span>
-          <div id="mv-notice-x" tabIndex="1" class="mv-x"></div>
-        </span>
-      </div>
-    </div>
     <div id="attribution"><div id="attribution-text"></div></div>
   </div>
 </body>
diff -pruN chromium-43.0.2357.18.orig/chrome/browser/resources/local_ntp/local_ntp.js chromium-43.0.2357.18/chrome/browser/resources/local_ntp/local_ntp.js
--- chromium-43.0.2357.18.orig/chrome/browser/resources/local_ntp/local_ntp.js	2015-04-15 05:55:03.000000000 +0200
+++ chromium-43.0.2357.18/chrome/browser/resources/local_ntp/local_ntp.js	2015-04-23 09:46:06.807900140 +0200
@@ -220,19 +220,19 @@ var fakeboxInputBehavior = NTP_DISPOSE_S
 
 
 /** @type {number} @const */
-var MAX_NUM_TILES_TO_SHOW = 8;
+var MAX_NUM_TILES_TO_SHOW = 0;
 
 
 /** @type {number} @const */
-var MIN_NUM_COLUMNS = 2;
+var MIN_NUM_COLUMNS = 0;
 
 
 /** @type {number} @const */
-var MAX_NUM_COLUMNS = 4;
+var MAX_NUM_COLUMNS = 0;
 
 
 /** @type {number} @const */
-var NUM_ROWS = 2;
+var NUM_ROWS = 0;
 
 
 /**
@@ -1094,7 +1094,6 @@ function handlePostMessage(event) {
  * Google-provided page.
  */
 function init() {
-  tilesContainer = $(IDS.TILES);
   notification = $(IDS.NOTIFICATION);
   attribution = $(IDS.ATTRIBUTION);
   ntpContents = $(IDS.NTP_CONTENTS);
@@ -1157,7 +1156,6 @@ function init() {
 
   ntpApiHandle = topLevelHandle.newTabPage;
   ntpApiHandle.onthemechange = onThemeChange;
-  ntpApiHandle.onmostvisitedchange = onMostVisitedChange;
 
   ntpApiHandle.oninputstart = onInputStart;
   ntpApiHandle.oninputcancel = restoreNtp;
diff -pruN chromium-43.0.2357.18.orig/chrome/browser/search/instant_service.cc chromium-43.0.2357.18/chrome/browser/search/instant_service.cc
--- chromium-43.0.2357.18.orig/chrome/browser/search/instant_service.cc	2015-04-15 05:55:04.000000000 +0200
+++ chromium-43.0.2357.18/chrome/browser/search/instant_service.cc	2015-04-23 09:46:56.303895906 +0200
@@ -126,9 +126,6 @@ InstantService::InstantService(Profile*
   // on Android.
 #if !defined(OS_ANDROID)
   content::URLDataSource::Add(profile_, new LocalNtpSource(profile_));
-  content::URLDataSource::Add(profile_, new ThumbnailSource(profile_, false));
-  content::URLDataSource::Add(profile_, new ThumbnailSource(profile_, true));
-  content::URLDataSource::Add(profile_, new ThumbnailListSource(profile_));
 #endif  // !defined(OS_ANDROID)
 
   favicon::FaviconService* favicon_service =
@@ -143,9 +140,6 @@ InstantService::InstantService(Profile*
       profile_, new FallbackIconSource(fallback_icon_service));
   content::URLDataSource::Add(
       profile_, new FaviconSource(profile_, FaviconSource::FAVICON));
-  content::URLDataSource::Add(profile_, new MostVisitedIframeSource());
-  content::URLDataSource::Add(
-      profile_, new suggestions::SuggestionsSource(profile_));
 }
 
 InstantService::~InstantService() {
diff -pruN chromium-43.0.2357.18.orig/chrome/browser/search/search.cc chromium-43.0.2357.18/chrome/browser/search/search.cc
--- chromium-43.0.2357.18.orig/chrome/browser/search/search.cc	2015-04-15 05:55:04.000000000 +0200
+++ chromium-43.0.2357.18/chrome/browser/search/search.cc	2015-04-23 09:46:06.808900140 +0200
@@ -268,7 +268,7 @@ NewTabURLState IsValidNewTabURL(Profile*
     return NEW_TAB_URL_INSECURE;
   if (!IsURLAllowedForSupervisedUser(new_tab_url, profile))
     return NEW_TAB_URL_BLOCKED;
-  return NEW_TAB_URL_VALID;
+  return NEW_TAB_URL_NOT_SET;
 }
 
 // Used to look up the URL to use for the New Tab page. Also tracks how we
diff -pruN chromium-43.0.2357.18.orig/chrome/browser/ui/search/search_ipc_router_policy_impl.cc chromium-43.0.2357.18/chrome/browser/ui/search/search_ipc_router_policy_impl.cc
--- chromium-43.0.2357.18.orig/chrome/browser/ui/search/search_ipc_router_policy_impl.cc	2015-04-15 05:55:04.000000000 +0200
+++ chromium-43.0.2357.18/chrome/browser/ui/search/search_ipc_router_policy_impl.cc	2015-04-23 09:46:06.808900140 +0200
@@ -23,7 +23,7 @@ SearchIPCRouterPolicyImpl::SearchIPCRout
 SearchIPCRouterPolicyImpl::~SearchIPCRouterPolicyImpl() {}
 
 bool SearchIPCRouterPolicyImpl::ShouldProcessSetVoiceSearchSupport() {
-  return true;
+  return false;
 }
 
 bool SearchIPCRouterPolicyImpl::ShouldProcessFocusOmnibox(bool is_active_tab) {
@@ -35,15 +35,15 @@ bool SearchIPCRouterPolicyImpl::ShouldPr
 }
 
 bool SearchIPCRouterPolicyImpl::ShouldProcessDeleteMostVisitedItem() {
-  return !is_incognito_ && chrome::IsInstantNTP(web_contents_);
+  return false;
 }
 
 bool SearchIPCRouterPolicyImpl::ShouldProcessUndoMostVisitedDeletion() {
-  return !is_incognito_ && chrome::IsInstantNTP(web_contents_);
+  return false;
 }
 
 bool SearchIPCRouterPolicyImpl::ShouldProcessUndoAllMostVisitedDeletions() {
-  return !is_incognito_ && chrome::IsInstantNTP(web_contents_);
+  return false;
 }
 
 bool SearchIPCRouterPolicyImpl::ShouldProcessLogEvent() {
@@ -89,7 +89,7 @@ bool SearchIPCRouterPolicyImpl::ShouldSe
 }
 
 bool SearchIPCRouterPolicyImpl::ShouldSendMostVisitedItems() {
-  return !is_incognito_ && chrome::IsInstantNTP(web_contents_);
+  return false;
 }
 
 bool SearchIPCRouterPolicyImpl::ShouldSendThemeBackgroundInfo() {
@@ -97,7 +97,7 @@ bool SearchIPCRouterPolicyImpl::ShouldSe
 }
 
 bool SearchIPCRouterPolicyImpl::ShouldSendToggleVoiceSearch() {
-  return true;
+  return false;
 }
 
 bool SearchIPCRouterPolicyImpl::ShouldSubmitQuery() {
diff -pruN chromium-43.0.2357.18.orig/chrome/browser/ui/webui/ntp/new_tab_ui.cc chromium-43.0.2357.18/chrome/browser/ui/webui/ntp/new_tab_ui.cc
--- chromium-43.0.2357.18.orig/chrome/browser/ui/webui/ntp/new_tab_ui.cc	2015-04-15 05:55:04.000000000 +0200
+++ chromium-43.0.2357.18/chrome/browser/ui/webui/ntp/new_tab_ui.cc	2015-04-23 09:46:06.809900139 +0200
@@ -96,7 +96,6 @@ NewTabUI::NewTabUI(content::WebUI* web_u
   if (!profile->IsOffTheRecord()) {
     web_ui->AddMessageHandler(new browser_sync::ForeignSessionHandler());
     web_ui->AddMessageHandler(new MetricsHandler());
-    web_ui->AddMessageHandler(new MostVisitedHandler());
     web_ui->AddMessageHandler(new RecentlyClosedTabsHandler());
     web_ui->AddMessageHandler(new FaviconWebUIHandler());
     web_ui->AddMessageHandler(new NewTabPageHandler());
@@ -228,7 +227,6 @@ void NewTabUI::RegisterProfilePrefs(
   NewTabPageHandler::RegisterProfilePrefs(registry);
   if (NewTabUI::IsDiscoveryInNTPEnabled())
     SuggestionsHandler::RegisterProfilePrefs(registry);
-  MostVisitedHandler::RegisterProfilePrefs(registry);
   browser_sync::ForeignSessionHandler::RegisterProfilePrefs(registry);
 }
 
diff -pruN chromium-43.0.2357.18.orig/components/history/core/browser/top_sites_backend.cc chromium-43.0.2357.18/components/history/core/browser/top_sites_backend.cc
--- chromium-43.0.2357.18.orig/components/history/core/browser/top_sites_backend.cc	2015-04-15 05:55:06.000000000 +0200
+++ chromium-43.0.2357.18/components/history/core/browser/top_sites_backend.cc	2015-04-23 09:46:06.809900139 +0200
@@ -21,56 +21,35 @@ namespace history {
 
 TopSitesBackend::TopSitesBackend(
     const scoped_refptr<base::SingleThreadTaskRunner>& db_task_runner)
-    : db_(new TopSitesDatabase()), db_task_runner_(db_task_runner) {
+    : db_(static_cast< TopSitesDatabase *> (NULL)), db_task_runner_(db_task_runner) {
   DCHECK(db_task_runner_);
 }
 
 void TopSitesBackend::Init(const base::FilePath& path) {
   db_path_ = path;
-  db_task_runner_->PostTask(
-      FROM_HERE, base::Bind(&TopSitesBackend::InitDBOnDBThread, this, path));
 }
 
 void TopSitesBackend::Shutdown() {
-  db_task_runner_->PostTask(
-      FROM_HERE, base::Bind(&TopSitesBackend::ShutdownDBOnDBThread, this));
 }
 
 void TopSitesBackend::GetMostVisitedThumbnails(
     const GetMostVisitedThumbnailsCallback& callback,
     base::CancelableTaskTracker* tracker) {
-  scoped_refptr<MostVisitedThumbnails> thumbnails = new MostVisitedThumbnails();
-  tracker->PostTaskAndReply(
-      db_task_runner_.get(), FROM_HERE,
-      base::Bind(&TopSitesBackend::GetMostVisitedThumbnailsOnDBThread, this,
-                 thumbnails),
-      base::Bind(callback, thumbnails));
 }
 
 void TopSitesBackend::UpdateTopSites(const TopSitesDelta& delta) {
-  db_task_runner_->PostTask(
-      FROM_HERE,
-      base::Bind(&TopSitesBackend::UpdateTopSitesOnDBThread, this, delta));
 }
 
 void TopSitesBackend::SetPageThumbnail(const MostVisitedURL& url,
                                        int url_rank,
                                        const Images& thumbnail) {
-  db_task_runner_->PostTask(
-      FROM_HERE, base::Bind(&TopSitesBackend::SetPageThumbnailOnDBThread, this,
-                            url, url_rank, thumbnail));
 }
 
 void TopSitesBackend::ResetDatabase() {
-  db_task_runner_->PostTask(
-      FROM_HERE,
-      base::Bind(&TopSitesBackend::ResetDatabaseOnDBThread, this, db_path_));
 }
 
 void TopSitesBackend::DoEmptyRequest(const base::Closure& reply,
                                      base::CancelableTaskTracker* tracker) {
-  tracker->PostTaskAndReply(db_task_runner_.get(), FROM_HERE,
-                            base::Bind(&base::DoNothing), reply);
 }
 
 TopSitesBackend::~TopSitesBackend() {
