diff -pruN chromium-47.0.2526.16.orig/chrome/browser/ui/views/layout_constants.cc chromium-47.0.2526.16/chrome/browser/ui/views/layout_constants.cc
--- chromium-47.0.2526.16.orig/chrome/browser/ui/views/layout_constants.cc	2015-07-23 22:09:20.000000000 +0200
+++ chromium-47.0.2526.16/chrome/browser/ui/views/layout_constants.cc	2015-07-29 23:28:01.114164014 +0200
@@ -6,11 +6,11 @@
   const int kLocationBarHorizontalPadding[] = {3, 6, 6};
   const int kLocationBarVerticalPadding[] = {2, 2, 2};
   const int kOmniboxDropdownBorderInterior[] = {6, 0, 0};
-  const int kOmniboxFontPixelSize[] = {16, 14, 14};
+  const int kOmniboxFontPixelSize[] = {12, 10, 10};
   const int kTabCloseButtonTrailingPaddingOverlap[] = {2, 2, 2};
   const int kTabFaviconTitleSpacing[] = {4, 4, 4};
-  const int kTabMaximumTitleWidth[] = {175, 175, 175};
-  const int kTabPinnedContentWidth[] = {25, 25, 25};
+  const int kTabMaximumTitleWidth[] = {125, 125, 125};
+  const int kTabPinnedContentWidth[] = {20, 20, 20};
 #if defined(OS_MACOSX)
   const int kTabstripTabOverlap[] = {19, 19, 19};
 #else
@@ -87,8 +87,8 @@ gfx::Insets GetLayoutInsets(LayoutInset
   const int kOmniboxDropdownIconPadding[] = {2, 4, 8};
   const int kOmniboxDropdownTextPadding[] = {3, 4, 8};
   const int kTabBottomPadding[] = {2, 2, 2};
-  const int kTabLeftPadding[] = {20, 20, 20};
-  const int kTabRightPadding[] = {20, 20, 20};
+  const int kTabLeftPadding[] = {18, 18, 18};
+  const int kTabRightPadding[] = {18, 18, 18};
   const int kTabTopPadding[] = {4, 4, 4};
   const int kToolbarBottomPadding[] = {5, 5, 5};
   const int kToolbarButtonPadding[] = {2, 6, 6};
diff -pruN chromium-45.0.2454.15.orig/chrome/browser/ui/views/frame/opaque_browser_frame_view.cc chromium-45.0.2454.15/chrome/browser/ui/views/frame/opaque_browser_frame_view.cc
--- chromium-45.0.2454.15.orig/chrome/browser/ui/views/frame/opaque_browser_frame_view.cc	2015-07-23 22:09:22.000000000 +0200
+++ chromium-45.0.2454.15/chrome/browser/ui/views/frame/opaque_browser_frame_view.cc	2015-07-29 23:26:14.620827222 +0200
@@ -78,7 +78,7 @@ const int kContentEdgeShadowThickness =
 
 #if !defined(OS_WIN)
 // The icon never shrinks below 16 px on a side.
-const int kIconMinimumSize = 16;
+const int kIconMinimumSize = 12;
 #endif
 
 #if defined(OS_LINUX) && !defined(OS_CHROMEOS)
diff -pruN chromium-45.0.2454.15.orig/chrome/browser/ui/views/frame/opaque_browser_frame_view_layout.cc chromium-45.0.2454.15/chrome/browser/ui/views/frame/opaque_browser_frame_view_layout.cc
--- chromium-45.0.2454.15.orig/chrome/browser/ui/views/frame/opaque_browser_frame_view_layout.cc	2015-07-23 22:09:22.000000000 +0200
+++ chromium-45.0.2454.15/chrome/browser/ui/views/frame/opaque_browser_frame_view_layout.cc	2015-07-29 23:26:14.620827222 +0200
@@ -33,7 +33,7 @@ const int kTitleLogoSpacing = 5;
 
 // The frame border is only visible in restored mode and is hardcoded to 4 px on
 // each side regardless of the system window border size.
-const int kFrameBorderThickness = 4;
+const int kFrameBorderThickness = 2;
 
 // The titlebar has a 2 px 3D edge along the top and bottom.
 const int kTitlebarTopAndBottomEdgeThickness = 2;
diff -pruN chromium-45.0.2454.15.orig/chrome/browser/ui/views/panels/panel_frame_view.cc chromium-45.0.2454.15/chrome/browser/ui/views/panels/panel_frame_view.cc
--- chromium-45.0.2454.15.orig/chrome/browser/ui/views/panels/panel_frame_view.cc	2015-07-23 22:09:22.000000000 +0200
+++ chromium-45.0.2454.15/chrome/browser/ui/views/panels/panel_frame_view.cc	2015-07-29 23:26:14.620827222 +0200
@@ -44,7 +44,7 @@ namespace {
 const int kNonAeroBorderThickness = 1;
 
 // The height and width in pixels of the icon.
-const int kIconSize = 16;
+const int kIconSize = 12;
 
 // The extra padding between the button and the top edge.
 const int kExtraPaddingBetweenButtonAndTop = 1;
diff -pruN chromium-45.0.2454.15.orig/chrome/browser/ui/views/toolbar/app_menu.cc chromium-45.0.2454.15/chrome/browser/ui/views/toolbar/app_menu.cc
--- chromium-45.0.2454.15.orig/chrome/browser/ui/views/toolbar/app_menu.cc	2015-07-23 22:09:22.000000000 +0200
+++ chromium-45.0.2454.15/chrome/browser/ui/views/toolbar/app_menu.cc	2015-07-29 23:26:14.624160555 +0200
@@ -77,7 +77,7 @@ using views::View;
 namespace {
 
 // Horizontal padding on the edges of the in-menu buttons.
-const int kHorizontalPadding = 15;
+const int kHorizontalPadding = 6;
 
 #if defined(OS_CHROMEOS)
 // Extra horizontal space to reserve for the fullscreen button.
diff -pruN chromium-45.0.2454.15.orig/ui/views/controls/menu/menu_config_aura.cc chromium-45.0.2454.15/ui/views/controls/menu/menu_config_aura.cc
--- chromium-45.0.2454.15.orig/ui/views/controls/menu/menu_config_aura.cc	2015-07-23 22:09:38.000000000 +0200
+++ chromium-45.0.2454.15/ui/views/controls/menu/menu_config_aura.cc	2015-07-29 23:26:14.624160555 +0200
@@ -32,18 +32,18 @@ void MenuConfig::Init(const ui::NativeTh
 
 void MenuConfig::InitAura(const ui::NativeTheme* theme) {
   submenu_horizontal_inset = 1;
-  arrow_to_edge_padding = 20;
+  //arrow_to_edge_padding = 12;
   ui::ResourceBundle& rb = ui::ResourceBundle::GetSharedInstance();
   arrow_width = rb.GetImageNamed(IDR_MENU_HIERARCHY_ARROW).Width();
   gfx::ImageSkia check = GetMenuCheckImage(false);
   check_height = check.height();
-  item_min_height = 29;
-  separator_spacing_height = 7;
-  separator_lower_height = 8;
-  separator_upper_height = 8;
-  label_to_arrow_padding = 20;
-  label_to_minor_text_padding = 20;
-  always_use_icon_to_label_padding = true;
+  item_min_height = 12;
+  //separator_spacing_height = 3;
+  //separator_lower_height = 4;
+  //separator_upper_height = 4;
+  //label_to_arrow_padding = 20;
+  //label_to_minor_text_padding = 20;
+  //always_use_icon_to_label_padding = true;
   align_arrow_and_shortcut = true;
   offset_context_menus = true;
   corner_radius = kMenuCornerRadiusForAura;
diff -pruN chromium-45.0.2454.15.orig/ui/views/window/custom_frame_view.cc chromium-45.0.2454.15/ui/views/window/custom_frame_view.cc
--- chromium-45.0.2454.15.orig/ui/views/window/custom_frame_view.cc	2015-07-23 22:09:38.000000000 +0200
+++ chromium-45.0.2454.15/ui/views/window/custom_frame_view.cc	2015-07-29 23:26:14.624160555 +0200
@@ -37,7 +37,7 @@ namespace {
 
 // The frame border is only visible in restored mode and is hardcoded to 4 px on
 // each side regardless of the system window border size.
-const int kFrameBorderThickness = 4;
+const int kFrameBorderThickness = 2;
 // In the window corners, the resize areas don't actually expand bigger, but the
 // 16 px at the end of each edge triggers diagonal resizing.
 const int kResizeAreaCornerSize = 16;
@@ -55,7 +55,7 @@ const int kTitleCaptionSpacing = 5;
 
 #if !defined(OS_WIN)
 // The icon never shrinks below 16 px on a side.
-const int kIconMinimumSize = 16;
+const int kIconMinimumSize = 12;
 #endif
 
 #if defined(OS_CHROMEOS)
