diff -pruN a/chromium-35.0.1916.47/chrome/browser/defaults.cc b/chromium-35.0.1916.47/chrome/browser/defaults.cc
--- a/chromium-35.0.1916.47/chrome/browser/defaults.cc	2014-04-16 22:32:38.000000000 +0200
+++ b/chromium-35.0.1916.47/chrome/browser/defaults.cc	2014-04-17 09:39:19.677952018 +0200
@@ -14,7 +14,7 @@ const bool kCanToggleSystemTitleBar = tr
 #endif
 #endif
 
-const int kOmniboxFontPixelSize = 16;
+const int kOmniboxFontPixelSize = 11;
 
 const bool kRestorePopups = false;
 
diff -pruN a/chromium-35.0.1916.47/chrome/browser/defaults.h b/chromium-35.0.1916.47/chrome/browser/defaults.h
--- a/chromium-35.0.1916.47/chrome/browser/defaults.h	2014-04-16 22:32:38.000000000 +0200
+++ b/chromium-35.0.1916.47/chrome/browser/defaults.h	2014-04-17 09:39:19.677952018 +0200
@@ -28,9 +28,9 @@ extern const int kOmniboxFontPixelSize;
 // all translation units that do this.
 #if defined(TOOLKIT_VIEWS)
 // Windows and Chrome OS have bigger shadows in the tab art.
-const int kMiniTabWidth = 64;
+const int kMiniTabWidth = 60;
 #else
-const int kMiniTabWidth = 56;
+const int kMiniTabWidth = 52;
 #endif
 
 // Should session restore restore popup windows?
diff -pruN a/chromium-35.0.1916.47/chrome/browser/ui/views/frame/opaque_browser_frame_view.cc b/chromium-35.0.1916.47/chrome/browser/ui/views/frame/opaque_browser_frame_view.cc
--- a/chromium-35.0.1916.47/chrome/browser/ui/views/frame/opaque_browser_frame_view.cc	2014-04-16 22:32:23.000000000 +0200
+++ b/chromium-35.0.1916.47/chrome/browser/ui/views/frame/opaque_browser_frame_view.cc	2014-04-17 09:39:19.677952018 +0200
@@ -75,7 +75,7 @@ const int kResizeAreaCornerSize = 16;
 const int kContentEdgeShadowThickness = 2;
 
 // The icon never shrinks below 16 px on a side.
-const int kIconMinimumSize = 16;
+const int kIconMinimumSize = 12;
 
 #if defined(OS_LINUX) && !defined(OS_CHROMEOS)
 // The number of pixels to move the frame background image upwards when using
diff -pruN a/chromium-35.0.1916.47/chrome/browser/ui/views/frame/opaque_browser_frame_view_layout.cc b/chromium-35.0.1916.47/chrome/browser/ui/views/frame/opaque_browser_frame_view_layout.cc
--- a/chromium-35.0.1916.47/chrome/browser/ui/views/frame/opaque_browser_frame_view_layout.cc	2014-04-16 22:32:24.000000000 +0200
+++ b/chromium-35.0.1916.47/chrome/browser/ui/views/frame/opaque_browser_frame_view_layout.cc	2014-04-17 09:39:19.678952018 +0200
@@ -33,7 +33,7 @@ const int kTitleLogoSpacing = 5;
 
 // The frame border is only visible in restored mode and is hardcoded to 4 px on
 // each side regardless of the system window border size.
-const int kFrameBorderThickness = 4;
+const int kFrameBorderThickness = 2;
 
 // The titlebar has a 2 px 3D edge along the top and bottom.
 const int kTitlebarTopAndBottomEdgeThickness = 2;
diff -pruN a/chromium-35.0.1916.47/chrome/browser/ui/views/panels/panel_frame_view.cc b/chromium-35.0.1916.47/chrome/browser/ui/views/panels/panel_frame_view.cc
--- a/chromium-35.0.1916.47/chrome/browser/ui/views/panels/panel_frame_view.cc	2014-04-16 22:32:23.000000000 +0200
+++ b/chromium-35.0.1916.47/chrome/browser/ui/views/panels/panel_frame_view.cc	2014-04-17 09:39:19.678952018 +0200
@@ -43,7 +43,7 @@ namespace {
 const int kNonAeroBorderThickness = 1;
 
 // The height and width in pixels of the icon.
-const int kIconSize = 16;
+const int kIconSize = 12;
 
 // The extra padding between the button and the top edge.
 const int kExtraPaddingBetweenButtonAndTop = 1;
diff -pruN a/chromium-35.0.1916.47/chrome/browser/ui/views/tabs/tab.cc b/chromium-35.0.1916.47/chrome/browser/ui/views/tabs/tab.cc
--- a/chromium-35.0.1916.47/chrome/browser/ui/views/tabs/tab.cc	2014-04-16 22:32:24.000000000 +0200
+++ b/chromium-35.0.1916.47/chrome/browser/ui/views/tabs/tab.cc	2014-04-17 09:39:19.679952018 +0200
@@ -178,7 +178,7 @@ static const int kFaviconTitleSpacing =
 static const int kTitleTextOffsetYAsh = 1;
 static const int kTitleTextOffsetY = 0;
 static const int kTitleCloseButtonSpacing = 3;
-static const int kStandardTitleWidth = 175;
+static const int kStandardTitleWidth = 125;
 // Additional vertical offset for close button relative to top of tab.
 // Ash needs this to match the text vertical position.
 static const int kCloseButtonVertFuzzAsh = 1;
diff -pruN a/chromium-35.0.1916.47/chrome/browser/ui/views/toolbar/wrench_menu.cc b/chromium-35.0.1916.47/chrome/browser/ui/views/toolbar/wrench_menu.cc
--- a/chromium-35.0.1916.47/chrome/browser/ui/views/toolbar/wrench_menu.cc	2014-04-16 22:32:24.000000000 +0200
+++ b/chromium-35.0.1916.47/chrome/browser/ui/views/toolbar/wrench_menu.cc	2014-04-17 09:39:19.679952018 +0200
@@ -74,10 +74,10 @@ namespace {
 // Horizontal padding on the edges of the buttons.
 const int kHorizontalPadding = 6;
 // Horizontal padding for a touch enabled menu.
-const int kHorizontalTouchPadding = 15;
+const int kHorizontalTouchPadding = 6;
 
 // Menu items which have embedded buttons should have this height in pixel.
-const int kMenuItemContainingButtonsHeight = 43;
+const int kMenuItemContainingButtonsHeight = 34;
 
 // Returns true if |command_id| identifies a bookmark menu item.
 bool IsBookmarkCommand(int command_id) {
diff -pruN a/chromium-35.0.1916.47/ui/views/controls/menu/menu_config_views.cc b/chromium-35.0.1916.47/ui/views/controls/menu/menu_config_views.cc
--- a/chromium-35.0.1916.47/ui/views/controls/menu/menu_config_aura.cc	2014-04-16 22:33:16.000000000 +0200
+++ b/chromium-35.0.1916.47/ui/views/controls/menu/menu_config_aura.cc	2014-04-17 09:39:19.680952018 +0200
@@ -31,18 +31,18 @@ void MenuConfig::Init(const ui::NativeTh
 
 void MenuConfig::InitAura(const ui::NativeTheme* theme) {
   submenu_horizontal_inset = 1;
-  arrow_to_edge_padding = 20;
+  //arrow_to_edge_padding = 12;
   ui::ResourceBundle& rb = ui::ResourceBundle::GetSharedInstance();
   arrow_width = rb.GetImageNamed(IDR_MENU_HIERARCHY_ARROW).Width();
   gfx::ImageSkia check = GetMenuCheckImage(false);
   check_height = check.height();
-  item_min_height = 29;
-  separator_spacing_height = 7;
-  separator_lower_height = 8;
-  separator_upper_height = 8;
-  label_to_arrow_padding = 20;
-  label_to_minor_text_padding = 20;
-  always_use_icon_to_label_padding = true;
+  item_min_height = 12;
+  //separator_spacing_height = 3;
+  //separator_lower_height = 4;
+  //separator_upper_height = 4;
+  //label_to_arrow_padding = 20;
+  //label_to_minor_text_padding = 20;
+  //always_use_icon_to_label_padding = true;
   align_arrow_and_shortcut = true;
   offset_context_menus = true;
   corner_radius = kMenuCornerRadiusForAura;
diff -pruN a/chromium-35.0.1916.47/ui/views/window/custom_frame_view.cc b/chromium-35.0.1916.47/ui/views/window/custom_frame_view.cc
--- a/chromium-35.0.1916.47/ui/views/window/custom_frame_view.cc	2014-04-16 22:33:17.000000000 +0200
+++ b/chromium-35.0.1916.47/ui/views/window/custom_frame_view.cc	2014-04-17 09:39:19.680952018 +0200
@@ -33,7 +33,7 @@ namespace {
 
 // The frame border is only visible in restored mode and is hardcoded to 4 px on
 // each side regardless of the system window border size.
-const int kFrameBorderThickness = 4;
+const int kFrameBorderThickness = 2;
 // In the window corners, the resize areas don't actually expand bigger, but the
 // 16 px at the end of each edge triggers diagonal resizing.
 const int kResizeAreaCornerSize = 16;
@@ -45,7 +45,7 @@ const int kTitlebarTopAndBottomEdgeThick
 // The icon is inset 2 px from the left frame border.
 const int kIconLeftSpacing = 2;
 // The icon never shrinks below 16 px on a side.
-const int kIconMinimumSize = 16;
+const int kIconMinimumSize = 12;
 // The space between the window icon and the title text.
 const int kTitleIconOffsetX = 4;
 // The space between the title text and the caption buttons.
