--- chromium-33.0.1750.115/build/common.gypi.orig	2014-02-21 11:53:43.916922963 +0100
+++ chromium-33.0.1750.115/build/common.gypi	2014-02-21 12:07:44.203951976 +0100
@@ -48,7 +48,7 @@
           # implementation. Certificate verification will in most cases be
           # handled by the OS. If OpenSSL's struct X509 is used to represent
           # certificates, use_openssl_certs must be set.
-          'use_openssl%': 0,
+          'use_openssl%': 0, # enabling breaks build
 
           # Typedef X509Certificate::OSCertHandle to OpenSSL's struct X509*.
           'use_openssl_certs%': 0,
@@ -397,7 +397,7 @@
       'enable_extensions%': 1,
 
       # Enable Google Now.
-      'enable_google_now%': 1,
+      'enable_google_now%': 0,
 
       # Enable printing support and UI. This variable is used to configure
       # which parts of printing will be built. 0 disables printing completely,
@@ -722,7 +722,7 @@
         ['OS=="android" or OS=="ios"', {
           'enable_captive_portal_detection%': 0,
         }, {
-          'enable_captive_portal_detection%': 1,
+          'enable_captive_portal_detection%': 0,
         }],
 
         # Enable Skia UI text drawing incrementally on different platforms.
