--- chromium-33.0.1750.115/build/common.gypi.orig	2014-02-21 11:53:43.916922963 +0100
+++ chromium-33.0.1750.115/build/common.gypi	2014-02-21 12:07:44.203951976 +0100
@@ -397,7 +397,7 @@
       'enable_extensions%': 1,
 
       # Enable Google Now.
-      'enable_google_now%': 1,
+      'enable_google_now%': 0,
 
       # Enable printing support and UI. This variable is used to configure
       # which parts of printing will be built. 0 disables printing completely,
