--- chrome/browser/password_manager/native_backend_kwallet_x.cc.orig	2014-07-20 13:34:40.774336139 +0200
+++ chrome/browser/password_manager/native_backend_kwallet_x.cc	2014-07-20 13:38:12.821004885 +0200
@@ -33,10 +33,10 @@ namespace {
 const char kKWalletFolder[] = "Chrome Form Data";
 
 // DBus service, path, and interface names for klauncher and kwalletd.
-const char kKWalletServiceName[] = "org.kde.kwalletd";
-const char kKWalletPath[] = "/modules/kwalletd";
+const char kKWalletServiceName[] = "org.kde.kwalletd5";
+const char kKWalletPath[] = "/modules/kwalletd5";
 const char kKWalletInterface[] = "org.kde.KWallet";
-const char kKLauncherServiceName[] = "org.kde.klauncher";
+const char kKLauncherServiceName[] = "org.kde.klauncher5";
 const char kKLauncherPath[] = "/KLauncher";
 const char kKLauncherInterface[] = "org.kde.KLauncher";
 
@@ -186,7 +186,7 @@ bool NativeBackendKWallet::StartKWalletd
                                "start_service_by_desktop_name");
   dbus::MessageWriter builder(&method_call);
   std::vector<std::string> empty;
-  builder.AppendString("kwalletd");     // serviceName
+  builder.AppendString("kwalletd5");    // serviceName
   builder.AppendArrayOfStrings(empty);  // urls
   builder.AppendArrayOfStrings(empty);  // envs
   builder.AppendString(std::string());  // startup_id
